react-app/src/components/Room.js

```diff
@@ -0,0 +1,19 @@
+import React from 'react';
+import { UncontrolledCollapse, Button, CardBody, Card } from 'reactstrap';
+
+const Room = (props) => (
+  <div>
+    <Button outline color="primary" size="lg" block id="toggler" style={{ marginBottom: '1rem' }}>
+      {props.roomName}
+    </Button>
+    <UncontrolledCollapse toggler="#toggler">
+      <Card>
+        <CardBody>
+          Placeholder TimeTable
+        </CardBody>
+      </Card>
+    </UncontrolledCollapse>
```

Comment:
The buttons that control the collapse appear to trigger all underlying components rather than individual components. I think what we want to achieve in terms of UI is that when a user clicks a button (say, room "305") only the underlying component containing timetable for room "305" is triggered, not all of the timetables get triggered.

We need to change that into a "controlled" collapse.